be/src/pipeline/pipeline_task.cpp

```diff
@@ -86,14 +76,21 @@ bool PipelineTask::has_dependency() {
 }
 
 Status PipelineTask::open() {
-    if (_sink) {
-        RETURN_IF_ERROR(_sink->open(_state));
-    }
+    auto st = Status::OK();
     for (auto& o : _operators) {
-        RETURN_IF_ERROR(o->open(_state));
+        if (st.is_blocked_by_rf()) {
+            o->open(_state);
```

Comment:
should RETURN_IF_ERROR, if some opeartor open failed, should as failed as quickly